### PR TITLE
Mermaid and kroki sphinx support

### PR DESCRIPTION
**Pull request by jschaul**
This adds kroki and mermaid support to this repository. Actual diagrams and documentation will follow in a follow-up PR.

## Checklist:

Please tick the following before making your PR:

* [x] I ran `make` on this branch to build the docs and there are **no WARNINGs**.
* [x] After running `make`, I looked at the output by opening build/html/index.html in a browser and I'm satisfied with the rendering.

### CODE REVIEW

File: diagrams/Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+.PHONY: compile
+compile:
+	mkdir -p ./tmp
+	for f in *.mmd; do NAME=$${f:0:-4}; ./mmdc -i $$f -o $$NAME.svg; convert $$NAME.svg ./tmp/$$NAME.pdf; done
+
+.PHONY: watch
+watch-%: compile
+	okular ./tmp/$(*).pdf &
```
Comment (mdimjasevic) [inline diagrams/Makefile:8]:
```suggestion
	xdg-open ./tmp/$(*).pdf &
```
Not everyone uses Okular as their preferred PDF viewer. By relying on `xdg-open` we leave it to the user's configuration to pick the PDF viewer of choice.

File: diagrams/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Diagrams with Mermaid
+
+* try them out on https://mermaid.live/
+* see syntax on https://mermaid-js.github.io/mermaid/#/sequenceDiagram
+* locally compile them with `./mmdc` to `svg`: [mmdc / mermaid-cli](https://github.com/mermaid-js/mermaid-cli)
+    * `npm install @mermaid-js/mermaid-cli`
+* see mermaid [integrations](https://mermaid-js.github.io/mermaid/#/./integrations)
+* type 'make watch-filename.mmd' if you have `mmdc` and `okular` and `entr` locally available for one option of a "save + autoreload" workflow.
```
Comment (mdimjasevic) [inline diagrams/README.md:8]: A reference to Okular should be changed if you agree with the proposal above.

File: diagrams/docs.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+WIP.
+
+Non-federation message sending
```
Comment (mdimjasevic) [inline diagrams/docs.md:3]: Perhaps some markdown to make it clear it's a start of a section?

File: diagrams/docs.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+Message sending happens in two steps.
+
+Step 1 is the synchronous request from client to backend, which gets a response as soon as the backend has stored the message in its database.
+
+Step 2 is asynchronous: the actual delivery of the message may happen directly via websocket if a client is online; it may get a push notification if it’s a phone and offline; or it may get nothing. If not online, clients will “catch up” by means of calling the /notifications  endpoint. It’s the client’s responsibility to store a “pointer”, the UUID of the last message they received/decrypted, and they say “give me everything newer than the last message I know about”. Message UUIDs are time-based, and as long as all the backend servers roughly agree on what time it is, message loss is rare.
+
+See this simplified flow diagram (click it!):
+
+Federation “M1” message sending
```
Comment (mdimjasevic) [inline diagrams/docs.md:13]: Ditto.

**Comment by jschaul** [comment]
@mdimjasevic thanks for the review; I fixed the okular issue, and will make a separate follow-up PR for the actual docs and actual diagrams.

File: diagrams/Makefile (hunk 1 above)
Comment (jschaul) [reply diagrams/Makefile]: yes, thanks! fixed.

**Review by mdimjasevic — APPROVED** [review]

**Pull request merged**